### ISSUE TITLE
Sign Up

### ISSUE THREAD

**Issue by Vedsaga** [Monofyi/Mobile#3]
1. Sign UP as owner :

![image](https://user-images.githubusercontent.com/59083528/125867567-d2b5eef3-f56f-42f1-b88b-9ae587126fe1.png)

2. Sign UP as worker :
![image](https://user-images.githubusercontent.com/59083528/125867600-69198f49-5962-4a25-b7df-22dc4339d6b6.png)

**Comment by Vedsaga** [comment]
@arun-muralidharan There is slightly changes in Sign-up UI.

![Screenshot_20210723_120716](https://user-images.githubusercontent.com/59083528/126745879-c74801ad-901e-49e2-bd26-87392452b31c.jpg)
![Screenshot_20210723_120750](https://user-images.githubusercontent.com/59083528/126745896-38861a5d-342e-40cc-a7be-22fd8f2a1c19.jpg)



**Most important part is that, Sign-up procedure is same for both owner and workers.**


In existing steps is no need to make any changes. We just need skip one step  which is when user select thier user-type as worker we will not seek owner references name during Sign-up. Which means user have to just select either they are worker or owner and once they click on Sign-up we will ask them to verify thier mail. Once that is done now they are ready to login. 

That's all what user have to while Sign-up. 

1. FILL ALL THE DETAILS 
2. SELECT USERTYPE
3. VERIFY THIER MAIL

**Comment by Vedsaga** [comment]
For time being, Email verification is not deployed that means we can skip that and can work on Sign-in.  Except Email Verification everything is already completed what was required for the Worker-Owner Sign-up.

**Comment by Vedsaga** [comment]
@arun-muralidharan @imssurya I am closing this issue. As everything is working fine in the apk. We will reopen this issue when Email Verification is deployed.

**Comment by Vedsaga** [comment]
**Email Verification is deployed. So, reopening the Sign-up** 
There are two steps to verify the mail one by clicking on link which is attached to thier mail.  Or by filling OTP. 

Steps to verify Email by filling OTP : 

1. Check email
![Screenshot_20210724_130158](https://user-images.githubusercontent.com/59083528/126861157-95190f09-c31c-4aa8-a15c-2686eeeebe71.jpg)

2. Enter OTP 
![Screenshot_20210724_130203](https://user-images.githubusercontent.com/59083528/126861227-86044645-4d3d-4f57-9ed0-bd8ae5945763.jpg)


3. Then comes the successful screen. 
![Screenshot_20210724_131242](https://user-images.githubusercontent.com/59083528/126861400-a449832b-d9bf-4a18-8f9c-692dc99900fd.jpg)



**2nd way** 

When user click on the link attached to the mail. We should redirect them to successful screen...

**Comment by arun-muralidharan** [comment]
@Vedsaga Should we also include a link within the verify page to resend OTP?

**Comment by Vedsaga** [comment]
Yaa,  I believe for Resend OTP there is separate API. So, it would be nice if that could be included.

**Comment by Vedsaga** [comment]
Sorry I just remembered that I missed that during designing the UI..  Please, could you include that

**Comment by Vedsaga** [comment]
**Added Resend OTP button in the UI..** 

![Screenshot_20210725_061350](https://user-images.githubusercontent.com/59083528/126884247-0cc30334-7c03-4387-a746-c61a06ba174f.jpg)

**Count - Down UI**
![Screenshot_20210725_061343](https://user-images.githubusercontent.com/59083528/126884253-a181f412-af91-433f-aedc-b0496e09d32f.jpg)

**Comment by arun-muralidharan** [comment]
#32 should close this, once it is verified and approved.

**Comment by Vedsaga** [comment]
So, when I open the app and try to sign up all the filed auto-filled. 
![Screenshot_20210728_083914](https://user-images.githubusercontent.com/59083528/127262763-11d404a0-4b46-4b39-ab12-6e400775b9a6.jpg)

​please could you fix it. Rest everything is working fine for Sign-up.

**Comment by Vedsaga** [comment]
There seems thier is problem while I was trying request for resend OTP. It's says Resend OTP Failed. 

![Screenshot_20210728_100736](https://user-images.githubusercontent.com/59083528/127264718-9766b5f0-b820-4d06-a234-02636e51235a.jpg)



Also, there seems to be another bug. So, when I requested for **Resend-OTP** and try to fill the OTP even when OTP is correct I am seeing multiple errors, screen shot attached below. 
![Screenshot_20210728_100910](https://user-images.githubusercontent.com/59083528/127264950-3aca8d8f-4a9c-498f-91c8-bd50433bcbed.jpg)

Screen shot of the valid OTP. 
![Screenshot_20210728_100915](https://user-images.githubusercontent.com/59083528/127264954-65859f5b-61c7-42dd-9ad9-469f32558884.jpg)

**Comment by Vedsaga** [comment]
I just noticed that, if I have already clicked on Resend OTP and again try to click on confim button then it show Resend OTzp failed. I have attached screen-recording. 
https://user-images.githubusercontent.com/59083528/127265341-df18c636-cb15-440e-9293-530e68650e5d.mp4

**Comment by arun-muralidharan** [comment]
That's odd. I'll take a look.

**Comment by arun-muralidharan** [comment]
@Vedsaga I found the bug for resend otp failing and not changing state; that is fixed. But I could not replicate the otp failing for a valid one. Maybe it was due to the state error. Please check after #34; if the issue persists and let me know.

**Comment by Vedsaga** [comment]
Thanks, sure will check. I felt that, if I try to register as worker and then try to verify the mail then it was not working may be that was the reason because the invalid OTP error occurred while I was trying to register as worker.. Sorr Owner everything was working..

**Comment by arun-muralidharan** [comment]
You are right; it doesn't work for worker. I just tested it. But this seems to be a backend issue. It didn't work outside the app (in Insomnia) either.

**Comment by Vedsaga** [comment]
> You are right; it doesn't work for worker. I just tested it. But this seems to be a backend issue. It didn't work outside the app (in Insomnia) either.

We can inform backend team. They can look into it and fix it.

**Comment by Vedsaga** [comment]
I guess, **Resend OTP error** occurring from backend API and once new API is deployed it would solve? 

![Screenshot_20210728_153857](https://user-images.githubusercontent.com/59083528/127305365-ce735c0d-0225-4deb-91a0-b9e559d9ed16.jpg)

**Comment by arun-muralidharan** [comment]
@Vedsaga Yes, we should test it after the changes. Resend and verify currently doesn't work for worker IDs.

**Comment by Vedsaga** [comment]
Everything working fine during Sign-up except the Email Verification error which occur for worker if he tries to verify after resend OTP for that new issue is separately opened #37

**Comment by Vedsaga** [comment]
Anna there seems to some issues while I am trying to sign up and if in Username fields if I pass more  then expected  `string` then it stuck at Sign-up process screen... Can please check this..  I ahev attach screen recording.. I guess I have passed `username` with string length more then 100+ not sure either the issue is from backend or frondend. 


https://user-images.githubusercontent.com/59083528/128500127-1f1262d9-8dcb-4396-97f5-72577107fdcd.mp4

**Comment by arun-muralidharan** [comment]
> Anna there seems to some issues while I am trying to sign up and if in Username fields if I pass more then expected `string` then it stuck at Sign-up process screen... Can please check this.. I ahev attach screen recording.. I guess I have passed `username` with string length more then 100+ not sure either the issue is from backend or frondend.

I'll check.
